### PR TITLE
fix funding

### PR DESCRIPTION
**Pull request by richmanbtc**
changes

- fix liquidation free funding (deleveraging)

### PR Reminders

Be ware of the followings

- [ ] implement deployment script for your changes in deployment repo
- [ ] add verification in hardhat simulation(000-prepare-simulation-check, 902-newMarket-check or 903-simulation-check) and system test(901-system-test) in deployment repo
- [ ] update change log
 
---

1. **Contract**: make sure the code follows our convention, ex: naming, explicit returns, etc.; if uncertain, discuss with others

2. **Test**: make sure tests can cover most normal and edge cases; if not, open follow-up tickets. Also, look out for failed tests and fix them!

3. **CHANGELOG.md**: update when external interfaces are changed

4. Workflow: 
    - Github: assign the pr to yourself; if pairing, can merge directly; else, assign someone to help review
    - Asana: assign the corresponding ticket to yourself and leave the pr link on it for easier follow-ups
    - Discord: if someone is mentioned in this pr, tag on Discord

### CODE REVIEW

File: contracts/PerpdexMarket.sol [hunk 1]
```diff
@@ -115,33 +121,54 @@ contract PerpdexMarket is IPerpdexMarket {
...
     }
 
-    function balanceToShare(uint256 balance) external view override returns (uint256) {
-        return balance.div(fundingInfo.balancePerShare);
+    function baseBalancePerShare() external view override returns (uint256) {
+        return poolInfo.baseBalancePerShare;
     }
 
     function _rebase() private {
         uint256 markPriceX96 = getMarkPriceX96();
```
Comment (takami-ml) [inline contracts/PerpdexMarket.sol:158]: should we consider the case when the liquidity is zero ? (e.g. initializing liquidity. poolInfo.base is zero, maybe?)

Comment (richmanbtc) [reply contracts/PerpdexMarket.sol:161]: Thank you. It needs to be considered. I want to fix it while writing a test
